pkg/querier/iterators/chunk_iterator.go

```diff
@@ -58,7 +62,26 @@ func (i *chunkIterator) At() (int64, float64) {
 	return i.cachedTime, i.cachedValue
 }
 
-func (i *chunkIterator) Next() bool {
+func (i *chunkIterator) AtHistogram() (int64, *histogram.Histogram) {
+	panic(errors.New("chunkIterator: AtHistogram not implemented"))
+}
+
+func (i *chunkIterator) AtFloatHistogram() (int64, *histogram.FloatHistogram) {
+	panic(errors.New("chunkIterator: AtFloatHistogram not implemented"))
+}
+
+func (i *chunkIterator) AtT() int64 {
+	if i.cacheValid {
+		return i.cachedTime
+	}
+
+	v := i.it.Value()
+	i.cachedTime, i.cachedValue = int64(v.Timestamp), float64(v.Value)
+	i.cacheValid = true
+	return i.cachedTime
```

Comment:
How about this
```go
t, _ := i.At()
return t
```

I also remember seeing something like this in the files that I have already seen (will be difficult to find that out now). So instead of duplicating almost all the At() code, you can do something like above at the other places as well if you come across it.